### ISSUE TITLE
[ML] Additional information in data frame analysis instrumentation

### ISSUE THREAD

**Issue by tveasey** [elastic/ml-cpp#1289]
1. Add in an estimation model size, i.e. the total number of nodes. This is useful for example if two hyperparameter choices have very different model sizes, but similar accuracy. In this case, the user may want to train with the parameters which result in smaller model. (This is a precursor to being able to target model size as part of the optimisation objective.)
2. The time take to train one forest with the parameters. Currently, we output the total time, but this doesn't account for the number of folds or if we stop early based on the validation loss. This will give someone an accurate estimation of how long retraining with the selected parameters overridden will take.
